src/tasks/FileUploadTask/FileObjectClasses/StreamUpload.ts

```diff
@@ -3,7 +3,29 @@ import { Readable } from "stream";
 import { GraphClientError } from "../../../GraphClientError";
 import { FileObject, SliceType } from "../../LargeFileUploadTask";
 import { Range } from "../Range";
+
+/**
+ * @interface
+ * Interface to store slice of a stream and range of the chunk.
+ * @property {Buffer} chunk - The slice of the stream
+ * @property {Range} range - The range of the slice
+ */
+interface ChunkRecord {
+	chunk: Buffer;
+	range: Range;
+}
+
+/**
+ * @class
+ * FileObject class for Readable Stream upload
+ */
 export class StreamUpload implements FileObject<Readable> {
+	/**
+	 * @private
+	 * Previous slice information.
+	 */
+	private previousChunk: ChunkRecord;
```

Comment:
This represents a cache of the last attempted upload slice. This should be used when resuming an upload.